pkg/controllers/managementuser/rbac/roletemplates/register.go

```diff
@@ -0,0 +1,13 @@
+package roletemplates
+
+import (
+	"context"
+
+	"github.com/rancher/rancher/pkg/types/config"
+)
+
+func Register(ctx context.Context, workload *config.UserContext) {
+	management := workload.Management.WithAgent("rbac-role-templates")
+
+	management.Management.RoleTemplates("").AddLifecycle(ctx, "cluster-roletemplate-handler", newRoleTemplateLifecycle(workload))
+}
```

Comment:
Conditional start based on the annotation `exp.cattle.io/aggregation: true`. A similar/negated logic should be added in its legacy counterpart.

```suggestion
package roletemplates

import (
	"context"

	v32 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
	"github.com/rancher/rancher/pkg/types/config"
	"k8s.io/apimachinery/pkg/runtime"
)

const featureFlagAnnotation = "exp.cattle.io/aggregation"

func Register(ctx context.Context, workload *config.UserContext) {
	// Only register if the condition is met
	starter := workload.DeferredStart(ctx, func(ctx context.Context) error {
		registerDeferred(ctx, workload)
		return nil
	})

	// Use a temporary handler to watch this specific Cluster object.
	// starter() needs to run just once, so using a child context allows cancelling once the condition is met,
	// effectively removing this temporary handler
	handlerCtx, stopHandler := context.WithCancel(ctx)

	clusterName := workload.ClusterName
	clusters := workload.Management.Management.Clusters("")
	clusters.AddHandler(handlerCtx, "certs-expiration-deferred", func(key string, obj *v32.Cluster) (runtime.Object, error) {
		if obj == nil || obj.Name != clusterName {
			return obj, nil
		}

		annotations := obj.GetAnnotations()
		if annotations != nil && annotations[featureFlagAnnotation] == "true" {
			if err := starter(); err != nil {
				return obj, err
			}
			stopHandler()
		}
		return obj, nil
	})

}

func registerDeferred(ctx context.Context, workload *config.UserContext) {
	management := workload.Management.WithAgent("rbac-role-templates")

	management.Management.RoleTemplates("").AddLifecycle(ctx, "cluster-roletemplate-handler", newRoleTemplateLifecycle(workload))
}
```